ReverseGeocode Does not work

At least in this case, using Google geocoder the result is a list of possible addresses, causing an exception to be raised in `GeocodeBase._add_location_to_signal`.

```
[2018-07-11T23:00:42.061600Z] NIO [ERROR] [svc.7f01f97f-cc54-49a3-9188-648af8311e6b] Unable to geocode location from query: (<lat>, <long>)
Traceback (most recent call last):
  File "C:\Users\Tom\nio\block_audit\blocks\geo\geocode_base.py", line 68, in _add_location_to_signal
    'address': location.address,
AttributeError: 'list' object has no attribute 'address'
```

--- a/geocode_base.py
+++ b/geocode_base.py
@@ -64,6 +64,8 @@ class GeocodeBase(Block):
                 self.logger.warning(
                     "No geocode loaction for query: {}".format(query))
                 return
+            if isinstance(location, list):
+                location = location[0]
             location_dict = {
                 'address': location.address,
                 'altitude': location.altitude,

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,2 @@
-geopy
+geopy~=1.14.0
 python-geohash

--- a/tests/test_distance_block.py
+++ b/tests/test_distance_block.py
@@ -69,7 +69,7 @@ class TestDistanceBlock(NIOBlockTestCase):
 
         self.assert_num_signals_notified(1)
         self.assertAlmostEqual(
-            self.last_notified[DEFAULT_TERMINAL][0].geodata['miles'], 537.1485,
+            self.last_notified[DEFAULT_TERMINAL][0].geodata['miles'], 536.998,
             2,
         )
 

--- a/tests/test_reverse_geocode_block.py
+++ b/tests/test_reverse_geocode_block.py
@@ -95,3 +95,22 @@ class TestReverseGeocodeBlock(NIOBlockTestCase):
         self.assert_num_signals_notified(1)
         self.assertEqual(
             self.last_notified[DEFAULT_TERMINAL][0].location, None)
+
+    def test_listed_results(self):
+        """Geocoder may return a list of results, the first one is used."""
+        blk = ReverseGeocode()
+        self.configure_block(blk, {})
+        blk._geolocator = MagicMock()
+        blk._geolocator.reverse.return_value = [
+            Location('foo', (1, 2), {"et": "cetera"}),
+            Location('bar', (3, 4), {"et": "alia"})]
+        blk.start()
+        blk.process_signals([Signal({"lat": 3, "lng": 14})])
+        blk.stop()
+        self.assertEqual(
+            self.last_notified[DEFAULT_TERMINAL][0].location['address'],
+            'foo')
+        self.assertEqual(
+            self.last_notified[DEFAULT_TERMINAL][0].location['latitude'], 1)
+        self.assertEqual(
+            self.last_notified[DEFAULT_TERMINAL][0].location['longitude'], 2)

